Remotion/Web/Development.Analyzers.IntegrationTests/RMWEB0003_TypeNotFromSameAssemblyWxeResourcePageStepAnalyzerTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Verifier = Remotion.Web.Development.Analyzers.IntegrationTests.CSharpAnalyzerVerifier<Web.Development.Analyzers.RMWEB0003TypeNotFromSameAssemblyWxeResourcePageStepAnalyzer>;
+
+namespace Remotion.Web.Development.Analyzers.IntegrationTests
+{
+  [TestFixture]
+  public class RMWEB0003TypeNotFromSameAssemblyWxeResourcePageStepAnalyzerTests
+  {
+    [Test]
+    public async Task TypeFromOtherAssembly_Invalid ()
```

Comment:
I found two more things: 
1. we need to detect usaes GetType(). If it's a lot of work, we can do it later. Using GetType will likely result in incorrect code once derived types are used. 
2. WxeResourceUserControlStep also needs to be covered. We can do this later if it's too much work.

Reply:
2) done

1) Detection of `.GetType` is a bit more complex that just checking for a `typeof(...)` expression since the `.GetType` is method call and you need to resolve everything to get determine if there is a problem. Right now, we only check for `typeof(xxx)` since that is straightforward, checking for something else gets difficult quickly as you need to check variables involved and on.

How about we have a general warning that we emit if you don't supply a value using a `typeof(xxx)`? This would reduce the complexity and requires consumers to check if they are affect. If they want they can even turn off this warning in the editorconfi

Reply:
Hmm...yeah, and someone who supplies a programmatically resolved type can just use `type.Assembly` instead of `type` to not have to suppress the warning.

Reply:
Actually, do we really need an analyzer for the type parameter? Initially the reason we wanted it was to prevent copy-paste errors. With the code fix this isn't a problem anymore...

Reply:
In the future, the ctor without the type parameter is going to the way of the Dodo. So new code will need to be written correctly from the start which will still produce copy&paste errors if one's not careful.